features/data_converter/failure/feature.go

```diff
@@ -0,0 +1,78 @@
+package failure
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"go.temporal.io/api/failure/v1"
+	historyProto "go.temporal.io/api/history/v1"
+	"go.temporal.io/sdk-features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+)
+
+var Feature = harness.Feature{
+	Workflows:   Workflow,
+	Activities:  FailureActivity,
+	CheckResult: CheckResult,
+	ClientOptions: client.Options{
+		FailureConverter: temporal.NewDefaultFailureConverter(temporal.DefaultFailureConverterOptions{
+			EncodeCommonAttributes: true,
+		}),
+	},
+}
+
+// run a workflow that calls an activity that will fail.
+func Workflow(ctx workflow.Context) error {
+	actCtx := workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+		ScheduleToCloseTimeout: 1 * time.Minute,
+		HeartbeatTimeout:       5 * time.Second,
+		RetryPolicy:            harness.RetryDisabled,
+	})
+	fut := workflow.ExecuteActivity(actCtx, FailureActivity)
+
+	_ = fut.Get(ctx, nil)
+	return nil
+}
+
+func FailureActivity(ctx context.Context) error {
+	return temporal.NewApplicationErrorWithCause("main error", "customType", errors.New("cause error"))
+}
+
+func CheckResult(ctx context.Context, runner *harness.Runner, run client.WorkflowRun) error {
+	if err := run.Get(ctx, nil); err != nil {
+		return err
+	}
+
+	history := runner.Client.GetWorkflowHistory(ctx, run.GetID(), "", false, 0)
```

Comment:
I think it'd be clearer to just return error out of the workflow and check the workflow error and causes instead of history. It's a clearer end-to-end test. (same for other langs)

Reply:
I'm fine with returning it from the workflow, but If I don't check the history how do I know if the message and stack trace got encoded ?

Reply:
Not sure what I was thinking back when I made this comment. I probably just wasn't paying attention that you assert the attributes.